src/commitment.rs

```diff
@@ -283,20 +266,23 @@ mod tests {
             let opening = commitment.open_at(point);
 
             // does evaluation match?
-            assert_eq!(point::to_u64(opening.value), value);
+            assert_eq!(opening.value.as_u64(), value);
 
             // does commitment match?
             let mut commitment_serialization = vec![0u8; 48];
             unsafe {
-                blst::blst_p1_compress(commitment_serialization.as_mut_ptr(), &commitment.element);
+                blst::blst_p1_compress(
+                    commitment_serialization.as_mut_ptr(),
+                    &commitment.element.point,
+                );
             }
             let expected_commitment_serialization = hex::decode(expected_commitment_hex).unwrap();
             assert_eq!(commitment_serialization, expected_commitment_serialization);
 
             // does proof match?
             let mut proof_serialization = vec![0u8; 48];
             unsafe {
-                blst::blst_p1_compress(proof_serialization.as_mut_ptr(), &opening.proof);
+                blst::blst_p1_compress(proof_serialization.as_mut_ptr(), &opening.proof.point);
```

Comment:
feeling like adding a serialization method to `oblast::PX` and then making `point` private again...

Reply:
you had actually already exposed the `compress` functions from `blst` so this was easy!